adapters/mqtt-vertx/src/main/java/org/eclipse/hono/adapter/mqtt/VertxBasedMqttProtocolAdapter.java

```diff
@@ -215,20 +215,26 @@ private void handleEndpointConnection(final MqttEndpoint endpoint) {
 
                         // check that MQTT client tries to publish on topic with device_id same as on connection
                         if (resource.getResourceId().equals(endpoint.clientIdentifier())) {
+                            // check credentials for valid authentication
+                            // TODO: how to determine the type? I would think: configurable for MQTT, hashed-password, preshared key maybe
```

Comment:
For now I agree: but I would think a config possibility would be helpful (and solves another point of the mentioned open points.

I think of something like (in different commit):
`hono
  mqtt:
    authentication: 
    - hashed-password
    - preshared-key`

Including the option to set it to `none`.
Currently only `hashed-password` would be supported and should be the default.
WDYT?